smt/expr.cpp

```diff
@@ -523,6 +523,18 @@ expr expr::lshr_exact(const expr &rhs) const {
   return (lshr(rhs) << rhs) == *this;
 }
 
+expr expr::bitreverse() const {
+  C();
+  auto nbits = bits();
+
+  expr res = mkUInt(0, nbits);
+  for (unsigned i = 0; i < nbits; i++) {
+    res = extract(i, i).zext(nbits - 1) | (res << mkUInt(1, nbits));
```

Comment:
Pranav, did you investigate any other encodings for this function to see which generate less trouble for the solver?

Reply:
I suggest extract of each bit + concat to append the bits together.

Reply:
i will take care of this in my next push

Reply:
changed to extract + concat